app/main/services/process_request_json.py

```diff
@@ -16,9 +16,34 @@ def process_values_for_matching(values):
     return values
 
 
+def _ensure_value_list(json_string_or_list):
+    if isinstance(json_string_or_list, list):
+        return json_string_or_list
+    else:
+        return [json_string_or_list]
+
+
 def convert_request_json_into_index_json(request_json):
     index_json = {}
 
+    for transformation in TRANSFORM_FIELDS:
+        # Each transformation can generate new field values in "target field" when
+        # certain values are present in "field". The example use case is when
+        # we are adding parent categories, whenever any one of their subcategories
+        # is present.
+        source_field = transformation['field']
+        target_field = transformation.get('target_field') or source_field
+
+        if source_field in request_json:
+            source_values = _ensure_value_list(request_json[source_field])
+            source_values_set = set(source_values)
+            target_values = _ensure_value_list(request_json[target_field])
```

Comment:
I suppose really if we have a 'transformation type' then the behaviour should be in some class (i.e. implementing a 'Transformation' interface). Which seems perfectly reasonable, but the sort of thing that should be refactored out once we have concrete examples?

For the yes-no-question to sub-category transformation, I think we potentially have enough here already: `field` just needs to be the original yes/no question, and `target_field` can be `'serviceCategories'`. (How that gets generated is another issue.)